website/content/docs/operations/garbage-collection.mdx

```diff
@@ -0,0 +1,212 @@
+  allocations. The client garbage collects older,
+  terminal allocations if adding new allocations would exceed the
+  `gc_max_allocs` limit.
+
+- User-initiated server garbage collection
+
+  When you initiate garbage collection on the server by running the `nomad
+  system gc` command, the clients collect all eligible allocations, bypassing
+  normal thresholds. Garbage collection first destroys all terminal allocations.
+  Then the process pops every remaining finished allocation and deletes it
+  within reason.
+
+Nomad does not continuously monitor disk or inode usage to trigger garbage
+collection. Instead, Nomad only checks disk and inode thresholds when one of the
+aforementioned triggers invokes the garbage collection process. The
+`gc_inode_usage_threshold` and `gc_disk_usage_threshold` values do not trigger
+garbage collection; rather, those values influence how the garbage collector
+behaves during a collection run.
+
+### Allocation selection
+
+When the garbage collection process runs, Nomad destroys as many finished
+allocations as needed to meet the resource thresholds. The client maintains a
+priority queue of terminal allocations ordered by the time they were marked
+finished, oldest first.
+
+The process repeatedly evicts allocations from the queue until the conditions
+are back within configured limits. Specifically, the garbage collection loop
+checks, in order:
+
+1. If disk usage exceeds `gc_disk_usage_threshold` value
+1. If inode usage exceeds `gc_inode_usage_threshold` value
+1. If the count of allocations exceeds `gc_max_allocs` value
+
+If any one of these conditions is true, the garbage collector selects the oldest
+finished allocation for removal.
+
+After deleting one allocation, the loop re-checks the metrics and continues
+removing the next-oldest allocation until all thresholds are satisfied or
+until there are no more terminal allocs. This means in a single run, the
+garbage collection removes multiple allocations back-to-back if the node was
+far over the limits. The evictions happen in termination-time order, which is
+oldest completed allocations first.
+
+If node's usage and allocation count are under the limits, a normal garbage
+collection cycle does not remove any allocations. In other words, periodic and
+event-driven garbage collection does not delete allocations just because they
+are finished. There has to be pressure or a limit reached. The exception is when
+an administrative command or server-side removal triggers client-side garbage
+collection. Aside from that forced scenario, the default behavior is
+threshold-driven: Nomad leaves allocations on disk until it needs to reclaim
+those allocations due to space, inode, or count limits being hit.
+
+### Task driver resources garbage collection
+
+A task driver does not have its own garbage collection process. Instead, the client garbage collection process triggers a task driver to clean up
+its resources when the allocation is terminal.
```

Comment:
Possible nitpick: this isn't quite accurate. The built-in `docker` driver _does_ have its own [`gc`](https://developer.hashicorp.com/nomad/docs/drivers/docker#gc) configuration block and although it's _mostly_ triggered by the client requesting task cleanup, it also has some internal loops that periodically cleanup. And third party task drivers _could_ have their own GC processes, but I don't know of any that do.